src/containers/GeetestCaptcha/index.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {
+    connect,
+    MapStateToProps,
+} from 'react-redux';
+import { initGeetest } from '../../helpers/geetest';
+import {
+    geetestCaptchaFetch,
+    GeetestCaptchaKeys,
+    GeetestCaptchaResponse,
+    RootState,
+    selectCaptchaKeys,
+    selectCurrentLanguage,
+} from '../../modules';
+
+interface OwnProps {
+    onSuccess?: (value?: GeetestCaptchaResponse) => void;
+}
+
+interface ReduxProps {
+    lang: string;
+    geetestCaptchaKeys?: GeetestCaptchaKeys;
+}
+
+interface DispatchProps {
+    geetestCaptchaFetch: typeof geetestCaptchaFetch;
+}
+
+type Props = ReduxProps & DispatchProps & OwnProps;
+
+class GeetestCaptchaComponent extends React.Component<Props> {
+
+    public constructor(props) {
+        super(props);
+        this.captchaContainerRef = React.createRef();
+    }
+
+    private captchaContainerRef;
+    private captcha;
+
+    public componentDidMount() {
+        this.props.geetestCaptchaFetch();
+    }
+
+    public componentWillReceiveProps(next: Props) {
+        if (this.props.geetestCaptchaKeys !== next.geetestCaptchaKeys && next.geetestCaptchaKeys !== undefined) {
```

Comment:
why do you check if `next.geetestCaptchaKeys !== undefined` if you can just check if `&& next.geetestCaptchaKeys`